narwhal/node/tests/reconfigure.rs

```diff
@@ -253,22 +268,62 @@ async fn restart() {
     // Listen to the outputs.
     let mut handles = Vec::new();
     for (tx, mut rx) in tx_clients.into_iter().zip(rx_nodes.into_iter()) {
+        let global_epoch = latest_observed_epoch.clone();
+        let execution_state = validators_execution_states.remove(0);
+
         handles.push(tokio::spawn(async move {
             let mut current_epoch = 0u64;
-
-            while let Some(epoch) = rx.recv().await {
-                println!("Received epoch {}", epoch);
-                if epoch == 5 {
-                    return;
-                }
-                if epoch > current_epoch {
-                    current_epoch = epoch;
-                    tx.send(current_epoch).await.unwrap();
+            static MAX_EPOCH: u64 = 10;
+
+            // Repeatedly send transactions.
+            let mut interval = interval(Duration::from_secs(3));
+            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+            tokio::pin!(interval);
+
+            loop {
+                tokio::select! {
+                    result = rx.recv() => {
+                        // channel broke - we need to exit the loop
```

Comment:
Nit: The wording "broke" is perhaps not ideal? 
> This method returns None if the channel has been closed and there are no remaining messages in the channel’s buffer
https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html

Reply:
good point, I'll refactor the comment